examples/step-74/step-laplace-FEIV.cc

```diff
@@ -0,0 +1,1245 @@
+      fe_v.get_function_hessians(solution, hessians);
+
+      std::vector<double> nu (n_q_points);
+      viscosity_function.value_list (fe_v.get_quadrature_points(), nu);
+      std::vector<double> rhs (n_q_points);
+      rhs_function.value_list (q_points, rhs);
+
+      double residual_norm_square = 0;
+      for (unsigned int point=0; point<n_q_points; ++point)
+        {
+          const double residual = rhs[point] + nu[point]*trace(hessians[point]);
+          residual_norm_square += residual*residual*JxW[point];
+        }
+        copy_data.value =cell->diameter()*std::sqrt(  residual_norm_square);
+    };
+
+    auto boundary_worker = [&] (const Iterator &cell, 
+                                const unsigned int &face_no, 
+                                ScratchData1<dim> &scratch_data, 
+                                CopyData &copy_data)
+    {
+      scratch_data.fe_interface_values.reinit(cell, face_no);
+      const FEFaceValuesBase<dim> &fe_fv =
+      scratch_data.fe_interface_values.get_fe_face_values(0);
+
+      const auto &q_points = fe_fv.get_quadrature_points();
+      const unsigned n_q_points = q_points.size();
+      
+      const std::vector<double> &JxW = fe_fv.get_JxW_values ();
+      
+      std::vector<double> nu (n_q_points);
+      viscosity_function.value_list (q_points, nu);
+
+      std::vector<double> g(n_q_points);
+      boundary_function.value_list (q_points, g);
+
+      std::vector<double> sol_u(n_q_points);
+      fe_fv.get_function_values(solution, sol_u);
+
+      const double degree = std::max(1.0, static_cast<double> (fe_fv.get_fe().degree));
+      const double extent1 = cell->extent_in_direction(GeometryInfo<dim>::unit_normal_direction[face_no]);
+      const double penalty = 4.0 * degree * (degree+1.0) / (extent1);
+     
+      double difference_norm_square = 0;
+      for (unsigned int point=0; point<q_points.size(); ++point)
+        {
+          const double diff= (g[point]-sol_u[point]);
+          difference_norm_square += diff*diff*JxW[point];
+        }
+        copy_data.value +=std::sqrt(penalty*difference_norm_square);
+    };
+
+    auto face_worker = [&] (const Iterator &cell, 
+                            const unsigned int &f, 
+                            const unsigned int &sf,
+                            const Iterator &ncell, 
+                            const unsigned int &nf,
+                            const unsigned int &nsf,
+                            ScratchData1<dim> &scratch_data,
+                            CopyData &copy_data)
+    {
+      FEInterfaceValues<dim> &fe_iv = scratch_data.fe_interface_values;
+      fe_iv.reinit(cell, f, sf, ncell, nf, nsf);  
+      
+      copy_data.face_data.emplace_back();
+      CopyDataFace &copy_data_face = copy_data.face_data.back();
+      
+      copy_data_face.cell_indices[0] = cell->active_cell_index();
+      copy_data_face.cell_indices[1] = ncell->active_cell_index();
```

Comment:
these should not be needed, I think if we replace the way we copy things out.

Reply:
> 
> 
> these should not be needed, I think if we replace the way we copy things out.

I am not sure how to implement this.